### PR TITLE
Adding PR template and workflow

### PR DESCRIPTION
**Pull request by spriya-m**

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+| -------------- |
+| |
+
+# Checklist:
+
+- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
+- [ ] I have verified that new and existing unit tests pass locally with my changes
+- [ ] I have not allowed coverage numbers to degenerate
+- [ ] I have maintained at least 90% code coverage
+- [ ] I have commented my code, particularly in hard-to-understand areas
```
Comment (shanmydell) [inline .github/pull_request_template.md:17]: - [ ] I have commented on my code, particularly in hard-to-understand areas

File: internal/wrappers/wrappers.go [hunk 2]
```diff
@@ -54,46 +63,59 @@ type ISCSILib interface {
...
 	return filepath.EvalSymlinks(path)
 }
 
+// OSWrapper contains implementation of LimitedOS interface
 type OSWrapper struct{}
 
+// OpenFile is a wrapper of os.OpenFile
 func (io *OSWrapper) OpenFile(name string, flag int, perm os.FileMode) (LimitedFile, error) {
-	return os.OpenFile(name, flag, perm)
+	return os.OpenFile(filepath.Clean(name), flag, perm) // #nosec G304
```
Comment (shanmydell) [inline internal/wrappers/wrappers.go:100]: After adding Filepath.clean, is nosec G304 requiredd?

**Review by shanmydell — COMMENTED** [review]
Pls address review comments

File: internal/mockhelper/mockhelper.go [hunk 3]
```diff
@@ -1,27 +1,40 @@
 package mockhelper
 
 import (
-	wrp "github.com/dell/gobrick/internal/wrappers"
 	"errors"
+	"os"
+
```
Comment (shanmydell) [inline internal/mockhelper/mockhelper.go:6]: This space is not necessary

File: internal/scsi/interface.go [hunk 4]
```diff
@@ -2,11 +2,13 @@ package scsi
 
 import (
 	"context"
+
```
Comment (shanmydell) [inline internal/scsi/interface.go:5]: Is this space required?

File: internal/mockhelper/mockhelper.go (hunk 3 above)
Comment (spriya-m) [reply internal/mockhelper/mockhelper.go:6]: Space was introduced by golint.

File: internal/scsi/interface.go (hunk 4 above)
Comment (spriya-m) [reply internal/scsi/interface.go:5]: Space was introduced by golint.

File: .github/pull_request_template.md (hunk 1 above)
Comment (spriya-m) [reply .github/pull_request_template.md:17]: this check is already added in PR template, anything in particular to be addressed?

Comment (shanmydell) [reply .github/pull_request_template.md:17]: @spriya-m : Please note the grammar check that has been corrected

Comment (spriya-m) [reply .github/pull_request_template.md]: ok, addressed the comment.

File: internal/wrappers/wrappers.go (hunk 2 above)
Comment (spriya-m) [reply internal/wrappers/wrappers.go:100]: Even with filepath.Clean(), G304 error is reported by gosec. So added nosec here, as few other repos have excluded this check from gosec runs.

Comment (shanmydell) [reply internal/wrappers/wrappers.go:100]: this is correct fix,but still gosec is reporting issue

**Review by shanmydell — APPROVED** [review]

**Review by harishp8889 — APPROVED** [review]

**Review by francis-nijay — APPROVED** [review]

**Pull request merged**